tests/bearlib/abstractions/LinterTest.py

```diff
@@ -856,3 +863,24 @@ def create_arguments(filename, file, config_file, some_value_B):
         self.assertEqual(create_arguments_mock.call_args[0][2][-5:], '.conf')
         generate_config_mock.assert_called_once_with(
             self.testfile2_path, self.testfile2_content, 124)
+
+    def test_capture_groups_warnings(self):
+        logger = logging.getLogger()
+        with self.assertLogs(logger, 'WARNING') as cm:
+            @linter('some-executable',
+                    use_stdout=True,
+                    output_format='regex',
+                    output_regex=r'(?P<not_supported_name>)\d+(\w+)')
+            class SomeBear:
+                pass
+
+        self.assertEqual(cm.output, [
+            'WARNING:root:SomeBear: Using unnecessary capturing groups '
+            'affects the performance of coala. '
+            "You should use '(?:<pattern>)' instead of "
+            "'(<pattern>)' for your regex.",
+
+            'WARNING:root:SomeBear: Superfluous capturing group'
+            " 'not_supported_name' used. Is this a typo? If not, consider"
+            " removing the capturing group to improve coala's "
+            'performance.'])
```

Comment:
please be consistent with strings, either end always with a space (which I prefer) or start everywhere with a space, don't mix both styles.